test/foundry-tests/PoolManagerReentrancyTest.sol

```diff
@@ -0,0 +1,167 @@
+    IPoolManager manager;
+
+    constructor(IPoolManager manager_) {
+        manager = manager_;
+    }
+
+    function main() external {
+        manager.lock("");
+    }
+
+    function checker0(uint256) external view {
+        assert(manager.locksLength() == 2);
+        assert(manager.lockIndex() == 1);
+        assert(manager.locksGetter(1).locker == msg.sender);
+        assert(manager.locksGetter(1).parentLockIndex == 0);
+    }
+
+    function checker1(uint256 depth) external view {
+        assert(manager.locksLength() == depth + 3);
+        assert(manager.lockIndex() == depth + 2);
+        assert(manager.locksGetter(depth + 2).locker == msg.sender);
+        if (depth == 0) assert(manager.locksGetter(depth + 2).parentLockIndex == 0);
+        else assert(manager.locksGetter(depth + 2).parentLockIndex == depth + 1);
+    }
+
+    function checker2(uint256) external view {
+        assert(manager.locksLength() == 5);
+        assert(manager.lockIndex() == 4);
+        assert(manager.locksGetter(4).locker == msg.sender);
+        assert(manager.locksGetter(4).parentLockIndex == 0);
+    }
+
+    function lockAcquired(bytes calldata) external returns (bytes memory) {
+        SimpleLinearLocker locker0 = new SimpleLinearLocker();
+        SimpleLinearLocker locker1 = new SimpleLinearLocker();
+        SimpleLinearLocker locker2 = new SimpleLinearLocker();
+
+        assert(manager.locksLength() == 1);
+        assert(manager.lockIndex() == 0);
+        assert(manager.locksGetter(0).locker == address(this));
+
+        locker0.main(manager, 0, this.checker0);
+        assert(manager.locksLength() == 2);
+        assert(manager.lockIndex() == 0);
+
+        locker1.main(manager, 1, this.checker1);
+        assert(manager.locksLength() == 4);
+        assert(manager.lockIndex() == 0);
+
+        locker2.main(manager, 0, this.checker2);
+        assert(manager.locksLength() == 5);
+        assert(manager.lockIndex() == 0);
+
+        return "";
+    }
+}
+
+contract PoolManagerReentrancyTest is Test, Deployers, TokenFixture {
+    PoolManager manager;
+
+    function setUp() public {
+        initializeTokens();
+        manager = Deployers.createFreshManager();
+    }
+
+    function testTokenLocker() public {
+        TokenLocker locker = new TokenLocker();
+        MockERC20(Currency.unwrap(currency0)).mint(address(locker), 1);
+        MockERC20(Currency.unwrap(currency0)).approve(address(locker), 1);
+        locker.main(manager, currency0, true);
+    }
+
+    function testTokenRevert() public {
+        TokenLocker locker = new TokenLocker();
+        MockERC20(Currency.unwrap(currency0)).mint(address(locker), 1);
+        MockERC20(Currency.unwrap(currency0)).approve(address(locker), 1);
+        vm.expectRevert(abi.encodeWithSelector(IPoolManager.CurrencyNotSettled.selector, address(locker)));
+        locker.main(manager, currency0, false);
+    }
+
+    function checker(uint256 depth) external {
```

Comment:
I know your tests are cute with short little names, but I find it takes some mental fortitude to wrap your head around them. Maybe something more descriptive like `assertionChecker` would help with readability

Reply:
changed the name :)

to be fair, writing reentrancy tests is pretty hard/wonky!